src/test/java/rx/internal/operators/OperatorObserveOnTest.java

```diff
@@ -523,14 +523,20 @@ public boolean hasNext() {
         assertEquals(7, generated.get());
     }
 
+    /**
+     * {@link BackpressureTests.testObserveOnWithSlowConsumer} also covers the case with default buffer size.
+     */
     @Test
     public void testQueueFullEmitsError() {
         final CountDownLatch latch = new CountDownLatch(1);
+        // randomize buffer size, note that underlaying implementations may be tuning the real size to a power of 2
+        // which can lead to unexpected results when adding excess capacity (e.g.: see ConcurrentCircularArrayQueue)
+        final int capacity = (int)Math.pow(2, new Random().nextInt(10));
```

Comment:
There is no need for randomized capacity. If you want to verify multiple sizes, loop over a power-of-2 set.